x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -140,20 +144,40 @@ export const useColumns = ({
   storage,
   id,
   defaultColumns,
+  initialBrowserFields,
 }: UseColumnsArgs) => {
   const [isBrowserFieldDataLoading, browserFields] = useFetchBrowserFieldCapabilities({
     featureIds,
+    initialBrowserFields,
   });
+
   const [columns, setColumns] = useState<EuiDataGridColumn[]>(storageAlertsTable.current.columns);
   const [isColumnsPopulated, setColumnsPopulated] = useState<boolean>(false);
 
+  const defaultColumnsRef = useRef<typeof defaultColumns>(defaultColumns);
+
+  const didDefaultColumnChanged = useMemo(
+    () => !isEqual(defaultColumns, defaultColumnsRef.current),
+    [defaultColumns]
+  );
+
+  useEffect(() => {
+    // if defaultColumns have changed, populate again
+    if (didDefaultColumnChanged) {
+      defaultColumnsRef.current = defaultColumns;
+      setColumns(storageAlertsTable.current.columns);
```

Comment:
if you do this it wouldn't persist on reload, wouldn't you want to setColumnsAndSave instead?

Reply:
@jcger 

So how I understood the logic is.. `persist` only runs if there is a modification such as `toggle` or `reset`. 

So in this case, it is by-default as sent by client, so it should not matter. If it was already in the storage. then `storageAlertsTable.current.columns` should have those value  already. does that make sense?